detectron2/data/datasets/pascal_voc.py

```diff
@@ -66,15 +66,13 @@ def load_voc_instances(dirname: str, split: str):
             bbox[0] -= 1.0
             bbox[1] -= 1.0
             instances.append(
-                {"category_id": CLASS_NAMES.index(cls), "bbox": bbox, "bbox_mode": BoxMode.XYXY_ABS}
+                {"category_id": class_names.index(cls), "bbox": bbox, "bbox_mode": BoxMode.XYXY_ABS}
             )
         r["annotations"] = instances
         dicts.append(r)
     return dicts
 
 
-def register_pascal_voc(name, dirname, split, year):
-    DatasetCatalog.register(name, lambda: load_voc_instances(dirname, split))
-    MetadataCatalog.get(name).set(
-        thing_classes=CLASS_NAMES, dirname=dirname, year=year, split=split
-    )
+def register_pascal_voc(name, dirname, split, class_names=CLASS_NAMES, **kwargs):
```

Comment:
Please keep `year` since it's required for evaluation.

Don't use mutable default: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments